<launch>
  <node pkg="nodelet" type="nodelet" name="z_filter"
        args="standalone pcl/PassThrough">
    <!-- <remap from="~input" to="/multisense/organized_image_points2_color" /> -->
    <remap from="~input" to="/multisense/resize_1_2/points" />
    <rosparam>
      filter_field_name: z
      filter_limit_max: 1.0
      keep_organized: true
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="organized_multi_plane_segmentation"
        name="multi_plane_estimate">
    <remap from="~input" to="z_filter/output"/>
  </node>
  
  <node pkg="nodelet" type="nodelet"
        name="plane_extraction"
        args="standalone jsk_pcl/MultiPlaneExtraction"
        output="screen">
    <remap from="~input" to="z_filter/output" />
    <remap from="~input_polygons" to="/plane_rejector/output_polygons" />
    <remap from="~input_coefficients" to="/plane_rejector/output_coefficients" />
    <rosparam>
      use_indices: false
    </rosparam>
  </node>
  <node pkg="nodelet" type="nodelet" name="plane_rejector"
        args="standalone jsk_pcl/PlaneRejector">
    <remap from="~input_polygons" to="/multi_plane_estimate/output_refined_polygon" />
    <remap from="~input_coefficients" to="/multi_plane_estimate/output_refined_coefficients" />
    <rosparam>
      processing_frame_id: multisense/head_root
      reference_axis: [0, 0, 1]
    </rosparam>
  </node>
  <node pkg="nodelet" type="nodelet" name="euclidean_clustering"
        args="standalone jsk_pcl/EuclideanClustering" output="screen">
    <remap from="~input" to="/plane_extraction/output" />
    <rosparam>
      tolerance: 0.02
      min_size: 100
      angle_thr: 0.5
    </rosparam>
  </node>

    <node pkg="nodelet" type="nodelet"
        name="cluster_decomposer"
        args="standalone jsk_pcl/ClusterPointIndicesDecomposer"
        output="screen">
    <remap from="~input" to="/plane_extraction/output" />
    <remap from="~target" to="/euclidean_clustering/output" />
    <remap from="~align_planes"
           to="/plane_rejector/output_polygons" />
    <remap from="~align_planes_coefficients"
           to="/plane_rejector/output_coefficients" />
    <rosparam>
      align_boxes: true
      use_pca: true
      publish_clouds: false
      publish_tf: false
      publish_tf: false
    </rosparam>
    </node>
    <node pkg="drc_task_common" type="standing_drill_detector"
          name="standing_drill_detector" output="screen">
      <remap from="~input" to="/plane_extraction/output" />
      <remap from="~input/box_array" to="/cluster_decomposer/boxes" />
      <remap from="~input/indices" to="/euclidean_clustering/output" />
    </node>
</launch>
